crates/fluvio-dataplane-protocol/src/batch.rs

```diff
@@ -740,8 +760,35 @@ mod test {
         assert_eq!(batch_created.last_offset_delta(), 2);
     }
 
+    #[test]
+    fn test_into_consumer_records_iter() {
+        let mut batch = Batch::from(vec![
+            Record::default(),
+            Record::default(),
+            Record::default(),
+        ]);
+
+        batch.header.first_timestamp = 1_500_000_000;
+        let partition_id = 1;
+
+        let consumer_records = batch
+            .into_consumer_records_iter(partition_id)
+            .collect::<Vec<ConsumerRecord>>();
+        assert_eq!(consumer_records.len(), 3);
+        assert_eq!(consumer_records[0].offset(), 0);
+        assert_eq!(consumer_records[1].offset(), 1);
+        assert_eq!(consumer_records[2].offset(), 2);
+
+        consumer_records.iter().for_each(|record| {
+            assert_eq!(record.timestamp(), 1_500_000_000);
+            assert_eq!(record.partition, partition_id);
+        });
+    }
```

Comment:
need to test if timestamp changes after putting new another batch

Reply:
Timestamp creation is done in MemoryBatch tests, this is just testing that this function works as expected